task/07-yield-tasks.js

```diff
@@ -126,7 +164,24 @@ function* breadthTraversalTree(root) {
  *   [ 1, 3, 5, ... ], [ -1 ] => [ -1, 1, 3, 5, ...]
  */
 function* mergeSortedSequences(source1, source2) {
-    throw new Error('Not implemented');
+    const src1 = source1()
+    const src2 = source2()
+    let srcValue1 = src1.next().value
+    let srcValue2 = src2.next().value
+
+    while(srcValue1 !== undefined || srcValue2 !== undefined) {
```

Comment:
```suggestion
    while(!src1.done() && !src1.done()) {
```
This check will be a bit more understable